ksqldb-engine/src/main/java/io/confluent/ksql/ddl/commands/CreateSourceFactory.java

```diff
@@ -229,7 +229,12 @@ private Formats buildFormats(
       final KsqlConfig ksqlConfig
   ) {
     final FormatInfo keyFormat = SourcePropertiesUtil.getKeyFormat(props, name);
+    // Validate format name and property
+    FormatFactory.of(keyFormat);
```

Comment:
Not sure exactly. Might be validated previously in other places like `DefaultSchemaInjector` and `SchemaRegisterInjector` implicitly. I don't fully recall why I added this, might be because I wrote some unit test for other PR previously and found incompatible properties can be set... 

Should be good to have a central place to do all validations explicitly so the preconditions are clear. 

I'm ok to remove this given this could be checked in other code path.